### DISCUSSION TITLE
Composer update

### DISCUSSION THREAD

**Discussion by dbarzin** [monarc-project/MonarcAppFO#498] [Q&A]
**Describe the bug**
Impossible to update the PHP libraries with PHP8.1

**To Reproduce**
1. Install Monarc in Ubuntu 22.04
2. install composer
3. go to Monac folder and try to "update composer"

**Expected behavior**
Get an updated set of PHP libraries

**Screenshots**
Not applicable

**Desktop (please complete the following information):**

| Questions                                | Answers
|-----------------------------------------|--------------------
| Type of issue                           | Improvement
| OS version (server)                 | Ubuntu 22.04
| OS version (client)                   | 
| PHP version                             | 8.1
| MONARC version / git hash     | MONARC 2.12.6
| Browser                                    | 

**Additional context**
composer.json need to be updated

$ composer update
Loading composer repositories with package information
Info from https://repo.packagist.org: #StandWithUkraine
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Root composer.json requires PHP extension ext-bcmath * but it is missing from your system. Install or enable PHP's bcmath extension.
  Problem 2
    - Root composer.json requires monarc/frontoffice ^2.12.6 -> satisfiable by monarc/frontoffice[v2.12.6].
    - monarc/frontoffice v2.12.6 requires ext-bcmath * -> it is missing from your system. Install or enable PHP's bcmath extension.
  Problem 3
    - ocramius/proxy-manager[2.13.0, ..., 2.13.x-dev] require php ~7.4.1 || ~8.0.0 -> your php version (8.1.2) does not satisfy that requirement.
    - ocramius/proxy-manager[2.14.0, ..., 2.15.x-dev] require php ~8.0.0 -> your php version (8.1.2) does not satisfy that requirement.
    - monarc/core v2.12.6 requires ocramius/proxy-manager ^2.13 -> satisfiable by ocramius/proxy-manager[2.13.0, ..., 2.15.x-dev].
    - Root composer.json requires monarc/core ^2.12.6 -> satisfiable by monarc/core[v2.12.6].
...

**Comment by ruslanbaidan** [comment]
You don't need to run the composer update command manually, it is done in update_all.sh script and there is an option presented there due to the one of the libraries limitation we can't fully switch to php8.1, so we ignore the version checkup
[here](https://github.com/monarc-project/MonarcAppFO/blob/master/scripts/update-all.sh#L67)

**Comment by dbarzin** [comment]
OK, thanks.
